pkg/deployment/inventory.go

```diff
@@ -183,198 +127,95 @@ func populateInventoryFromIPAM(
 	host.Vars["dns_search_domains"] = dnsSearchDomains
 }
 
-// getAnsibleUser returns the string value from the template unless it is set in the node
-func getAnsibleUser(instance *dataplanev1.OpenStackDataPlaneNodeSet, nodeName string) string {
-	if instance.Spec.Nodes[nodeName].Ansible.AnsibleUser != "" {
-		return instance.Spec.Nodes[nodeName].Ansible.AnsibleUser
-	}
-	return instance.Spec.NodeTemplate.Ansible.AnsibleUser
-}
-
-// getAnsiblePort returns the string value from the template unless it is set in the node
-func getAnsiblePort(instance *dataplanev1.OpenStackDataPlaneNodeSet, nodeName string) string {
-	if instance.Spec.Nodes[nodeName].Ansible.AnsiblePort > 0 {
-		return strconv.Itoa(instance.Spec.Nodes[nodeName].Ansible.AnsiblePort)
-	}
-	return strconv.Itoa(instance.Spec.NodeTemplate.Ansible.AnsiblePort)
-}
+func resolveGroupAnsibleVars(template *dataplanev1.NodeTemplate, group *ansible.Group,
+	defaultImages dataplanev1.DataplaneAnsibleImageDefaults) error {
+	groupVars := make(map[string]interface{})
 
-// getAnsibleManagementNetwork returns the string value from the template unless it is set in the node
-func getAnsibleManagementNetwork(
-	instance *dataplanev1.OpenStackDataPlaneNodeSet,
-	nodeName string) string {
-	if instance.Spec.Nodes[nodeName].ManagementNetwork != "" {
-		return instance.Spec.Nodes[nodeName].ManagementNetwork
+	if template.Ansible.AnsibleUser != "" {
+		groupVars["ansible_user"] = template.Ansible.AnsibleUser
```

Comment:
This will also address James' comment:
https://github.com/openstack-k8s-operators/dataplane-operator/pull/431#pullrequestreview-1644623434

Adding it to the group and allowing member hosts to default to it.